src/app/app-routing.module.ts

```diff
@@ -25,20 +25,21 @@ const appRoutes: Routes = [
         component: DeviceComponent
       },
       {
-        path: 'devices/:id', 
+        path: 'devices/:id',
         component: DeviceListComponent
       },
       {
         path: 'newlocation/:id',
         component: NewLocationComponent
       }
-    ] 
+    ]
   },
   { path: '**', component: PageNotFoundComponent }
 ];
 
 @NgModule({
   imports: [
+    RouterModule.forRoot(appRoutes, { }),
```

Comment:
http://stackoverflow.com/questions/36101756/angular2-routing-with-hashtag-to-page-anchor